python/tvm/driver/tvmc/compiler.py

```diff
@@ -0,0 +1,280 @@
+
+
+def drive_compile(args):
+    """ Invoke tvmc.compiler module with command line arguments
+
+    Parameters
+    ----------
+    args: argparse.Namespace
+        Arguments from command line parser.
+
+    Returns
+    --------
+    int
+        Zero if successfully completed
+
+    """
+
+    graph, lib, params, dumps = compile_model(
+        args.FILE,
+        args.target,
+        args.dump_code,
+        "",
+        args.model_format,
+        args.tuning_records,
+        args.tensor_layout,
+    )
+
+    if dumps:
+        save_dumps(args.output, dumps)
+
+    save_module(args.output, graph, lib, params, args.cross_compiler)
+    return 0
+
+
+def compile_model(
+        path,
+        target,
+        dump_code=None,
+        target_host=None,
+        model_format=None,
+        tuning_records=None,
+        alter_layout=None,
+):
+    """Compile a model from a supported framework into a TVM module.
+
+    This function takes a union of the arguments of both frontends.load_model
+    and compiler.compile_relay. The resulting TVM module can be executed using
+    the graph runtime.
+
+    Parameters
+    ----------
+    path: str
+        Path to a file
+    target : str
+        The target for which to compile. Can be a plain string or
+        a path.
+    dump_code : list, optional
+        Dump the generated code for the specified source types, on
+        the requested target.
+    target_host : str, optional
+        The target of the host machine if host-side code
+        needs to be generated.
+    model_format: str, optional
+        A string representing a name of a frontend to be used
+    tuning_records: str, optional
+        Name of the file produced by the tuning to be used during
+        compilation.
+    alter_layout: str, optional
+        The layout to convert the graph to. Note, the convert layout
+        pass doesn't currently guarantee the whole of the graph will
+        be converted to the chosen layout.
+
+    Returns
+    -------
+    graph : str
+        A JSON-serialized TVM execution graph.
+    lib : tvm.module.Module
+        A TVM module containing the compiled functions.
+    params : dict
+        The parameters (weights) for the TVM module.
+    dumps : dict
+        Dictionary containing the dumps specified.
+
+    """
+    dump_code = [x.strip() for x in  dump_code.split(',')] if dump_code else None
+    mod, params = frontends.load_model(path, model_format)
+
+    if alter_layout:
+        mod = common.convert_graph_layout(mod, alter_layout)
+
+    if os.path.exists(str(target)):
+        with open(target) as target_file:
+            logging.info("using target input from file: %s", target)
+            target = "".join(target_file.readlines())
+
+    # TODO(@leandron) We don't have an API to collect a list of supported
+    #       targets yet
+    logging.debug("creating target from input: %s", target)
+    tvm_target = tvm.target.create(target)
```

Comment:
I needed to revert this, as using `tvm.target.Target(target)` does not provide the features like JSON loading, etc.

Reply:
`create` is going to be deprecated so it's better to use `Target`. In fact, all uses of `create` in TVM code base have been changed to `Target` already.

`Target` doesn't provide JSON loading by purpose but recommends the following usage:
```python
with open('target.json', 'r') as fp:
  target = tvm.target.Target(fp.read())
```
As can be seen, you don't have to load a JSON file to a Python dict before creating a Target, because it directly works for JSON string.

Reply:
Understood - I'll update accordingly